app/models/product.rb

```diff
@@ -1,20 +1,47 @@
 class Product < ApplicationRecord
+  PERMALINK_REGEX = /[[:alnum:]]+/
+  validates_with ImageUrlValidator, attributes: [:image_url], if: ->{ image_url.present? }
+
   has_many :line_items
   has_many :orders, through: :line_items
   before_destroy :ensure_not_referenced_by_any_line_item
 
-  validates :title, :description, :image_url, presence: true
-  validates :price, numericality: { greater_than_or_equal_to: 0.01 }
+  validates :title, :description, :image_url, :price, presence: true
+
+  validates :price, numericality: { greater_than_or_equal_to: 0.01 }, if: :price_present?
+  validates_comparison_of :price, greater_than: :discount_price, message: "should be greater than discount price", if: [:price_present?, :discount_price_present?]
```

Comment:
Use validation helper.
Lets add this validation using Custom Validator as well, as required

Reply:
Done